server/daemon.py

```diff
@@ -194,6 +197,24 @@ def processor(result):
             return await self._send(payload, processor)
         return []
 
+    async def _is_rpc_available(self, method):
+        available = self._available_rpcs.get(method, None)
+        if available is None:
+            payload = {'method': method, 'id': self.next_req_id()}
+            data = json.dumps(payload)
+            result = await self._send_data(data)
+            if not isinstance(result, tuple):
```

Comment:
The general approach is what I had in mind.  I don't like the duplication here though.  Why not just call self._send_single like self.estimatefee does, and catch the thrown exception when not available?

Reply:
Well as far as I can see there is no exception to catch. If we call `self._send_single`, that calls `self._send`, and we reach L138. Log shows: `ERROR:Daemon:HTTP error code 404: Not Found  Retrying occasionally...`

Reply:
I see.  It's a shame bitcoind's JSON implementation is so bad, this isn't the only example.  It should be returning a JSON not HTTP error.  Perhaps the core loop should translate that error code to a DeamonError exception.

I'll take a closer look when I'm back.

Reply:
I can't test for several days myself but perhaps adding 400 to the list of error codes at line 100 for resp.status will have the effect of throwing the error.  Can you try that?

Reply:
It seems like bitcoind uses HTTP error codes independently of the JSON, and sends the JSON too.

Reply:
Looks like this will work. Will have something soon.